streamlit/streamlit_code_backup_1_02_(10_4_22).py

```diff
@@ -0,0 +1,639 @@
+# -*- coding: utf-8 -*-
+"""Streamlit Code Backup 1.02 (10/4/22)
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1gk4FpXHhs5H4v6RYh2-0SUbAfebNSgzy
+"""
+
+# In order to connect via another device, http://192.168.7.157:(localhostid)
+# localhostid can be found in the search bar once the app is opened on the main server computer
+
+# Current Bugs: None!
+# To-do: Automate downloads from the drive
+# Add map of roughly where each sensor is located
+# Refine Code (Add loops and such for excessive if statements)
+# Use st.select_slider to change the visible parts of the graph (Unviable, conflicts with custom_x/y too much and would require and entire rework of the code)
+# Add a 'heat map' of where light pollution is higher than others
+# Add calendar of what days have data for each sensor
+
+# Imports
+import pandas
+import matplotlib.pyplot as plt
+import numpy as np
+from numpy import random as r
+import csv
+import streamlit as st
+import plotly.express as px
+from datetime import datetime
+from datetime import timedelta
+from PIL import Image
+
+# Page setup
+st.set_page_config(
+    page_title="LENSS Sensor Plotting",
+    page_icon="🌠",
+    initial_sidebar_state="expanded",
+)
+
+
+# Login Page, only allows access once the correct password is entered
+with st.form("Login"):
+    input_password = st.text_input("Please Input Password", "")
+    submitted = st.form_submit_button("Submit")
+
+if input_password == "Y3rke5*":
```

Comment:
You should update this password to something else within the next few days. Since it is in this public repository, we can assume that is is no longer secure.